doc/api/process.md

```diff
@@ -1684,6 +1684,107 @@ the [`'warning'` event][process_warning] and the
 [`emitWarning()` method][process_emit_warning] for more information about this
 flag's behavior.
 
+## `process.parseArgs([argv[, options]])`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `argv` {Array<string>|Object} (Optional) Array of argument strings; defaults
+  to [`process.argv.slice(2)`](process_argv). If an Object, the default is used,
+  and this parameter is considered to be the `options` parameter.
+* `options` {Object} (Optional) The `options` parameter, if present, is an
+  object supporting the following property:
+  * `expectsValue` {Array<string>|string} (Optional) One or more argument
+    strings which _expect a value_ when present
+* Returns: {Object} An object having properties corresponding to parsed Options
```

Comment:
why not just return `{ options: { ... }, positionals: [...] }`? it gets rid of the awkward `_` handling and is imo much cleaner than magic properties.

Reply:
I'm not _opposed_ to this, but it is based on userland conventions.  _Personally_ I would prefer the magic over nested properties, but that's just me (as a potential consumer of this API).

Reply:
Yeah I've seen the userland libraries. I'd like to think we can do better :smile:

Reply:
I think I'm with Gus here, userland convention originated before deconstructors, so having a `_` property could be considered better than returning ` options: { ... }, positionals: [...] }`. The explicit object return is, IMO, more intuitive and user-friendly:

```js
const { options, positionals } = processArgs()
```

`_` might be somewhat familiar to users used to yargs, commander etc, but it won't be intuitive for new users.

Reply:
I think that `const { options, positionals } = processArgs()` is a pretty elegant compromise, I never loved `_` 😆